.github/workflows/codespell.yml

```diff
@@ -0,0 +1,22 @@
+---
+name: Codespell
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  codespell:
+    name: Check for spelling errors
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Codespell
```

Comment:
Would you consider adding a comment here, reminding a developer that the configuration is in the `.codespellrc` file?

This may be me being pedantic, so I won't block if you disagree / feel it's excessive.

Reply:
```suggestion
      - name: Codespell
      # uses configuration within .codespellrc file
```